### ISSUE TITLE
[Bug report] Catalog backend name handle the renaming of catalog

### ISSUE THREAD

**Issue by jerqi** [apache/gravitino#4718]
### Version

main branch

### Describe what's wrong

Currenlty, Iceberg jdbc backend will use catalog name as default backend name. In my understanding, it will trigger the bug if I rename the catalog.

### Error message and/or stacktrace

From the code.
```
catalogname
    String catalogName = properties.get("catalog-name");
    jdbcProperties.put(
        "iceberg.jdbc-catalog.catalog-name",
        properties.getOrDefault(IcebergConstants.CATALOG_BACKEND_NAME, catalogName));
```

### How to reproduce

I have questions when I read the code

**Comment by jerqi** [comment]
@FANNG1 Could you take a look?

**Comment by FANNG1** [comment]
thanks for reporting, I'll take it

**Comment by FANNG1** [comment]
1. The default value of CATALOG_BACKEND_NAME should be set by Gravitino server, the clients(Spark, Trino, Flink) should not care about it.
2. Compared to catalog id , I'd like to use origin `jdbc` as the default catalog name, as Iceberg REST server doesn't have something like catalog id.

@diqiu50 @jerqi @jerryshao WDYT?

**Comment by diqiu50** [comment]
What are the problems with using jdbc as the default catalog name?

**Comment by FANNG1** [comment]
> What are the problems with using jdbc as the default catalog name?

The two catalogs with the same JDBC URI will share the same table.

**Comment by diqiu50** [comment]
It feels like this is not a reasonable usage.

**Comment by diqiu50** [comment]
If the user wants to use one MySQL, at least the database name for Iceberg catalogs should not be the same

**Comment by FANNG1** [comment]
database is included in the `uri` for JDBC backend.

**Comment by diqiu50** [comment]
Why do two Iceberg catalogs need to use the same database? Do users expect these two catalogs to be able to see each other's tables?

**Comment by FANNG1** [comment]
> Why do two Iceberg catalogs need to use the same database? 

Maybe used for test? I'm not sure

> Do users expect these two catalogs to be able to see each other's tables?

I'm not sure, What do you think?

**Comment by diqiu50** [comment]
I think it maybe used for test.

**Comment by FANNG1** [comment]
@diqiu50 , could you summarize your point about the proposed changes?

**Comment by diqiu50** [comment]
I think that if the user sets the `catalog.backend.name`, it should use the user's setting; otherwise, it should use the default
